geomstats/_backend/tensorflow/__init__.py

```diff
@@ -498,9 +500,30 @@ def array_from_sparse(indices, data, target_shape):
     a : array, shape=target_shape
         Array of zeros with specified values assigned to specified indices.
     """
-    return _tf.sparse.to_dense(
-        _tf.sparse.reorder(_tf.SparseTensor(indices, data, target_shape))
-    )
+    if type(data) is list:
+        data = _tf.convert_to_tensor(_np.array(data))
+    data_type = data.dtype
+    if data_type is _tf.dtypes.complex64 or data_type is _tf.dtypes.complex128:
```

Comment:
Please, tell me if you see a way to avoid special cases for complex numbers. 
For now, if have modified my code using: 
if data_type in [_tf.dtypes.complex64, _tf.dtypes.complex128]:

Reply:
The problem is that I did not succeed to use _tf.sparse.to_dense() with complex matrices. (I have an error message.)
I can write:
    data = array(data)
    data_type = data.dtype
    sparse_data = _tf.sparse.reorder(_tf.SparseTensor(indices, _tf.convert_to_tensor(data, data_type), target_shape))
But then I cannot use _tf.sparse.to_dense(sparse_data) --> I have an error message. Any ideas?